integration/examples/hot-reload/skaffold.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: skaffold/v1alpha4
+kind: Config
+build:
+  artifacts:
+  - image: gcr.io/k8s-skaffold/node-example
+    context: node
+    sync:
+      '*.js': .
```

Comment:
Shouldn't we avoid user data as keys ?

Reply:
Why is that?

Reply:
I should have justified my comment indeed.
Coming from k8s it is unusual to have user data as keys; it surprised me.

Digging into k8s design doc, we have:
https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#lists-of-named-subobjects-preferred-over-maps
There are whole discussions linked from there, notably about
- readability (mostly in case of subobjects though)
- merging definitions, which may or may not be of concert for skaffold

It lists exceptions to this though: "pure maps [...] as opposed to sets of subobjects".

I suppose if we later want to enrich the format with subobjects it would make things harder with the current format.

I am not (yet) familiar with all the skaffold format, I don't know if there are other cases like that, I couln't find any.

So this is just to open the discussion. Should I move that into an issue?

Reply:
Thanks for the link. That actually makes a lot of sense. Please open an issue!  I think it’s a good suggestion

Reply:
here we go: #1180